src/server.cc

```diff
@@ -250,33 +250,59 @@ InferenceServer::Stop(const bool force)
     LOG_INFO << "Waiting for in-flight requests to complete.";
   }
 
-  Status status = model_repository_manager_->UnloadAllModels();
+  Status status = model_repository_manager_->StopAllModels();
   if (!status.IsOk()) {
     LOG_ERROR << status.Message();
   }
 
   // Wait for all in-flight non-inference requests to complete and all
   // loaded models to unload, or for the exit timeout to expire.
   uint32_t exit_timeout_iters = exit_timeout_secs_;
-
+  bool unloading_model = false;
   while (true) {
-    const auto& live_models = model_repository_manager_->LiveModelStates();
-
-    LOG_INFO << "Timeout " << exit_timeout_iters << ": Found "
-             << live_models.size() << " live models and "
-             << inflight_request_counter_
-             << " in-flight non-inference requests";
-    if (LOG_VERBOSE_IS_ON(1)) {
-      for (const auto& m : live_models) {
-        for (const auto& v : m.second) {
-          LOG_VERBOSE(1) << m.first << " v" << v.first << ": "
-                         << ModelReadyStateString(v.second.first);
+    if (!unloading_model) {
+      // Check if all in-flight inference requests / sequences are completed
+      const auto& inflight_status = model_repository_manager_->InflightStatus();
+      LOG_INFO << "Timeout " << exit_timeout_iters << ": Found "
+               << inflight_status.size()
+               << " model versions that have in-flight inferences";
+      if (LOG_VERBOSE_IS_ON(1)) {
+        for (const auto& inflight : inflight_status) {
+          LOG_VERBOSE(1) << std::get<0>(inflight) << " v"
```

Comment:
Do we want to include more information when logging the inflight requests here? For example, "Model std::get<0>(inflight) (version std::get<1>(inflight)) has std::get<2>(inflight) in-flight inference request(s)"